src/exec.js

```diff
@@ -104,8 +105,11 @@ function execSync(cmd, opts, pipe) {
     code = parseInt(fs.readFileSync(codeFile, 'utf8'), 10);
   }
 
-  var stdout = fs.readFileSync(stdoutFile, 'utf8');
-  var stderr = fs.readFileSync(stderrFile, 'utf8');
+  // fs.readFileSync uses buffer encoding by default, so set
+  // the encoding only if opts.encoding wasn't set to 'buffer'
+  var encoding = opts.encoding !== 'buffer' ? opts.encoding : null;
```

Comment:
That won't work, because the `'buffer'` encoding is only valid for the `child_process` methods:
```
> fs.readFileSync('./package.json', 'buffer')
Error: Unknown encoding: buffer
    at assertEncoding (fs.js:88:11)
    at Object.fs.readFileSync (fs.js:394:3)
    at repl:1:4
    at REPLServer.defaultEval (repl.js:272:27)
    at bound (domain.js:287:14)
    at REPLServer.runBound [as eval] (domain.js:300:12)
    at REPLServer.<anonymous> (repl.js:441:12)
    at emitOne (events.js:82:20)
    at REPLServer.emit (events.js:169:7)
    at REPLServer.Interface._onLine (readline.js:203:10)
```
If you need to get a buffer out of the `fs` methods, `encoding` needs to be `null` or `undefined`.